tests/console/consoletest_finish.pm

```diff
@@ -36,6 +36,13 @@ sub run {
     save_screenshot();
 
     systemctl 'unmask packagekit.service';
+    # On s390x sometimes the vnc will still there and the next select_console
+    # will create another vnc. This will make the OpenQA have 2 vnc sessions at
+    # same time. We'd cleanup the previous one and setup the new one.
+    if (!check_var("DESKTOP", "textmode") && check_var('ARCH', 's390x')) {
+        my $pid = script_output("ps -aux | grep Xvnc | grep -v grep | awk \'{print \$2}\'");
+        script_run("kill -9 $pid") if ($pid ne "");
+    }
```

Comment:
-kill :display#
    This kills a VNC desktop previously started with vncserver. It does this by killing the Xvnc process, whose process ID is stored in the file "$HOME/.vnc/host:display#.pid". The -kill option ignores anything preceding the first colon (":") in the display argument. Thus, you can invoke "vncserver -kill $DISPLAY", for example at the end of your xstartup file after a particular application exits. 

Huajian, I suggest you used the "vncserver -kill :display"

Reply:
vncserver -list can show the current display ids

Reply:
will try

Reply:
my $pid = script_output("vncserver -list | grep ^: |awk '{print \$1}''");
script_run("vncserver -kill $pid") if ($pid ne "");

Reply:
Hi Richard,
I've tried and it's not work, cause in this case the tty of this process is "?" and we can't get the pid by this normal way. thanks

Reply:
Sure,  then no problem from my side. LGTM